src/System.IO/tests/BufferedStream/BufferedStream.FakeStream.cs

```diff
@@ -0,0 +1,115 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+
+namespace BufferedStreamTests
+{
+    /// <summary>
+    /// An unnecessary buffer around a MemoryStream to ensure correctness of the underlying stream functions.
+    /// 
+    /// FakeStream is used as an indirection because BufferedStream prevents creation around a MemoryStream.
```

Comment:
This shouldn't be necessary once the Contract.Assert for MemoryStream is removed.